src/recipe/test/update-recipe.service.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { MongooseModule, getModelToken } from '@nestjs/mongoose';
+import { Test, TestingModule } from '@nestjs/testing';
+import { Model } from 'mongoose';
+
+import { closeConnections, rootMongooseTestModule } from '../../mock/db.mock';
+import { Recipe, RecipeDocument, RecipeSchema } from '../recipe.schema';
+import { RecipeService } from '../recipe.service';
+import { mockId, recipeMock } from './mock/recipe.mock';
+
+describe('RecipeService.update()', () => {
+  let recipeService: RecipeService;
+  let recipeModel: Model<RecipeDocument>;
+  let recipe: RecipeDocument;
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [rootMongooseTestModule(), MongooseModule.forFeature([{ name: Recipe.name, schema: RecipeSchema }])],
+      providers: [RecipeService],
+    }).compile();
+
+    recipeService = module.get(RecipeService);
+    recipeModel = module.get(getModelToken(Recipe.name));
+    recipe = new recipeModel(recipeMock());
+    await recipe.save();
+  });
+
+  afterAll(async () => {
+    await closeConnections();
+  });
+
+  it('should be defined', () => {
```

Comment:
imo you do not have to check if you correctly "imported" recipeService. It will throw error if it's undefined on call